torch/testing/_internal/common_utils.py

```diff
@@ -1264,6 +1292,26 @@ def assertObjectIn(self, obj: Any, iterable: Iterable[Any]) -> None:
                 return
         raise AssertionError("object not found in iterable")
 
+    def assertRaises(self, expected_exception, *args, **kwargs):
```

Comment:
Add comments here and for assertRaisesRegex() below on why the override is needed.

I can appreciate why we may need an override, but do we really need to subclass unittest.case._AssertRaisesContext? Seems like the overrides should give us the flexibility not to?